app/dashboard/src/pages/authentication/Login.tsx

```diff
@@ -33,119 +33,92 @@ import TextLink from '#/components/TextLink'
 /** A form for users to log in. */
 export default function Login() {
   const location = router.useLocation()
-  const { signInWithGoogle, signInWithGitHub, signInWithPassword } = authProvider.useAuth()
-  const { getText } = textProvider.useText()
-
+  const auth = useAuth()
+  const { getText } = useText()
   const query = new URLSearchParams(location.search)
   const initialEmail = query.get('email')
-
-  const [email, setEmail] = React.useState(initialEmail ?? '')
-  const [password, setPassword] = React.useState('')
-  const [isSubmitting, setIsSubmitting] = React.useState(false)
-  const shouldReportValidityRef = React.useRef(true)
-  const formRef = React.useRef<HTMLFormElement>(null)
-
-  const localBackend = backendProvider.useLocalBackend()
+  const localBackend = useLocalBackend()
   const supportsOffline = localBackend != null
 
+  const signInWithGoogleMutation = useMutation({ mutationFn: auth.signInWithGoogle })
+  const signInWithGitHubMutation = useMutation({ mutationFn: auth.signInWithGitHub })
+  const signInWithPasswordMutation = useMutation({
+    mutationFn: ({ email, password }: { email: string; password: string }) =>
+      auth.signInWithPassword(email, password),
+  })
+  const signInWithGoogle = signInWithGoogleMutation.mutateAsync
+  const signInWithGitHub = signInWithGitHubMutation.mutateAsync
+  const signInWithPassword = signInWithPasswordMutation.mutateAsync
```

Comment:
Not sure it's a great idea to use mutations for login. Since we send raw passwords, it might be possible to get that password from the mutation cache later after logging in.

Reply:
good point... i think they may (?) have been required before some refactor? but the other pages don't use mutations so no idea why i added them here